test/transform.js

```diff
@@ -327,3 +327,28 @@ check(
 
   , 'gitlab.com'
 )
+
+// check that the asteriskFlag successfully uses '* ' instead of '- '
+check(    
+    [ '# My Module'
+    , 'Some text here'
+    , '## API'
+    , '### Method One'
+    , 'works like this'
+    , '### Method Two'
+    , '#### Main Usage'
+    , 'some main usage here'
+    ].join('\n')
+  , [ '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*\n\n'
+    , '* [My Module](#my-module)\n'
+    ,   '  * [API](#api)\n'
+    ,     '    * [Method One](#method-one)\n'
+    ,     '    * [Method Two](#method-two)\n'
+    ,         '      * [Main Usage](#main-usage)\n\n\n'
+    ].join('')
+  , undefined
+  , undefined
+  , undefined
+  , undefined
+  , true // set asteriskFlag which is passed to transform() to true
+)
```

Comment:
Thanks for the note and explanatory link. I'm about to add a commit that will add the newline, and will respond to the rest of suggestions below.